components/match2/wikis/fifa/match_legacy.lua

```diff
@@ -0,0 +1,88 @@
+---
+-- @Liquipedia
+-- wiki=fifa
+-- page=Module:Match/Legacy
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local MatchLegacy = {}
+
+local json = require('Module:Json')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local UNKNOWNREASON_DEFAULT_LOSS = 'L'
+
+function MatchLegacy.storeMatch(match2, options)
+	local match, doStore = MatchLegacy.convertParameters(match2)
+
+	if options.storeMatch1 and doStore then
+		return mw.ext.LiquipediaDB.lpdb_match(
+			'legacymatch_' .. match2.match2id,
+			match
+		)
+	end
+end
+
+function MatchLegacy.convertParameters(match2)
```

Comment:
```suggestion
function MatchLegacy.convertParameters(match2)
	local match = Table.filterByKey(match2, function(key) return not String.startsWith(key, 'match2') end)

	local opponents = match2.match2opponents
	if #opponents ~= 2 or opponents[1].type ~= opponents[2].type or
		(opponents[1].type ~= Opponent.solo and opponents[1].type ~= Opponent.team)
	then return nil, false end

	match.staticid = match2.match2id
	match.extradata = Json.parseIfString(match.extradata) or {}
	match.mode = opponent1.type

	local handleOpponent = function(opponentIndex)
		local prefix = 'opponent' .. opponentIndex
		local opponent = opponents[opponentIndex]
		local players = opponent.match2players

		match[prefix .. 'score'] = (tonumber(opponent.score) or 0) >= 0 and opponent.score or 0

		if opponent.type == Opponent.tem then
			match[prefix] = String.isNotEmpty(opponent.name) and opponent.name:gsub('_', ' ') or 'TBD'
			return
		end

		--due to above filtering we have opponent.type == Opponent.solo
		local player = players[1] or {}
		match[prefix] = player.name and player.name:gsub('_', ' ') or nil
		match[prefix .. 'flag'] = player.flag
		match.extradata[prefix .. 'name'] = player.displayname
	end

	handleOpponent(1)
	handleOpponent(2)

	if match.resulttype == 'default' then
		match.resulttype = string.upper(match.walkover or '')
		if match.resulttype == UNKNOWNREASON_DEFAULT_LOSS then
			--needs to be converted because in match1 storage it was marked this way
		match.resulttype = 'unk'
		end
		match.walkover = match.winner
	end

	match.extradata.bestof = match2.bestof ~= 0 and tostring(match2.bestof) or ''
	match.extradata = json.stringify(match.extradata)

	return match, true
end
```

written on mobile so please test!

Reply:
were this to be added without replacing any other lines? (seeing there were some similarities to the doStore thing and if I paste all this by replacing only line 28, keeping all doStore stuff an <eof> error will return)

Reply:
replace the whole function

Reply:
> replace the whole function

Alright, It currently shows this 
![image](https://github.com/Liquipedia/Lua-Modules/assets/88981446/e5ce4ed9-14bd-42a2-bdc4-39a6b571f73e)

This was the revision I replaced, I assume the whole function goes down to **return match, dostore** line so I replaced whole of it https://liquipedia.net/fifa/index.php?title=Module%3AMatch%2FLegacy%2Fdev&type=revision&diff=201476&oldid=201475

Reply:
https://liquipedia.net/fifa/index.php?title=Module%3AMatch%2FLegacy%2Fdev&type=revision&diff=201493&oldid=201476